src/os_win/os_dir.c

```diff
@@ -59,30 +59,41 @@ __directory_list_worker(WT_FILE_SYSTEM *file_system,
 		WT_ERR(__wt_map_windows_error(windows_error));
 	}
 
-	count = 0;
-	do {
+	for (count = 0;;) {
 		/*
 		 * Skip . and ..
 		 */
 		if (wcscmp(finddata.cFileName, L".") == 0 ||
 		    wcscmp(finddata.cFileName, L"..") == 0)
-			continue;
+			goto skip;
 
 		/* The list of files is optionally filtered by a prefix. */
 		if (prefix != NULL &&
 		    wcsncmp(finddata.cFileName, prefix_wide->data,
 			prefix_widelen) != 0)
-			continue;
+			goto skip;
 
 		WT_ERR(__wt_realloc_def(
 		    session, &dirallocsz, count + 1, &entries));
-
 		WT_ERR(__wt_to_utf8_string(
 		    session, finddata.cFileName, &file_utf8));
 		WT_ERR(__wt_strdup(session, file_utf8->data, &entries[count]));
 		++count;
 		__wt_scr_free(session, &file_utf8);
-	} while (!single && FindNextFileW(findhandle, &finddata) != 0);
+
+		if (single)
+			break;
+
+skip:		if (FindNextFileW(findhandle, &finddata) != 0)
+			continue;
+		windows_error = __wt_getlasterror();
+		if (windows_error == ERROR_NO_MORE_FILES)
+			break;
+		__wt_errx(session,
+		    "%s: directory-list: FindNextFileW: %s",
+		    pathbuf->data, __wt_formatmessage(session, windows_error));
```

Comment:
Use `prefix` instead of `pathbuf->data` in the error message. `pathbuf->data` is `wchar_t` in this case.

Reply:
After a closer look, my comment is wrong